features/nanostack/mbed-mesh-api/source/wisun_tasklet.c

```diff
@@ -681,3 +704,93 @@ int wisun_tasklet_set_trusted_certificate(uint8_t *cert, uint16_t cert_len)
     }
     return wisun_tasklet_store_certificate_data(cert, cert_len, NULL, 0, false, false, true);
 }
+
+int wisun_tasklet_statistics_start(void)
+{
+    statistics_started = true;
+
+    if (statistics == NULL) {
+        statistics = ns_dyn_mem_alloc(sizeof(wisun_statistics_t));
+    }
+    if (statistics == NULL) {
+        return -1;
+    }
+    memset(statistics, 0, sizeof(wisun_statistics_t));
+
+    wisun_tasklet_statistics_do_start();
+
+    return 0;
+}
+
+static void wisun_tasklet_statistics_do_start(void)
+{
+    if (!wisun_tasklet_data_ptr || wisun_tasklet_data_ptr->network_interface_id < 0 || !mac_api) {
+        return;
+    }
+
+    protocol_stats_start(&statistics->nwk_stats);
+    ns_sw_mac_statistics_start(mac_api, &statistics->mac_statistics);
+    ns_sw_mac_phy_statistics_start(mac_api, &statistics->phy_statistics);
+    ws_statistics_start(wisun_tasklet_data_ptr->network_interface_id, &statistics->ws_statistics);
+    wisun_tasklet_statistics_fhss_do_start();
+
+}
+
+static void wisun_tasklet_statistics_fhss_do_start(void)
+{
+    if (!fhss_api) {
+        fhss_api = ns_sw_mac_get_fhss_api(mac_api);
+        if (fhss_api) {
+            ns_fhss_statistics_start(fhss_api, &statistics->fhss_statistics);
+        }
+    }
+}
+
+int wisun_tasklet_statistics_stop(void)
+{
+    statistics_started = false;
+
+    protocol_stats_stop();
+    if (wisun_tasklet_data_ptr) {
+        ws_statistics_stop(wisun_tasklet_data_ptr->network_interface_id);
+    }
+    return 0;
+}
+
+int wisun_tasklet_statistics_nw_read(mesh_nw_statistics_t *stats)
+{
+    if (!statistics) {
+        return -1;
+    }
+    memcpy(stats, &statistics->nwk_stats, sizeof(nwk_stats_t));
+    stats->asynch_tx_count = statistics->ws_statistics.asynch_tx_count;
+    stats->asynch_rx_count = statistics->ws_statistics.asynch_rx_count;
+    return 0;
+}
+
+int wisun_tasklet_statistics_phy_read(mesh_phy_statistics_t *stats)
+{
+    if (!statistics) {
+        return -1;
+    }
+    memcpy(stats, &statistics->phy_statistics, sizeof(phy_rf_statistics_s));
```

Comment:
nervous about to use that memcpy like at other functions. If structure name is same we can do just *stats = statistic->why_statistic;

Reply:
corrected